rerun_cpp/src/rerun/archetypes/asset3d_ext.cpp

```diff
@@ -1,17 +1,22 @@
 #include <algorithm>
-#include <filesystem> // TODO(#3991): Should not leak into public header.
-#include <fstream>    // TODO(#3991): Should not leak into public header.
+#include <fstream>
 #include <string>
 
 #include "asset3d.hpp"
 
-//#define EDIT_EXTENSION
+// It's undefined behavior to pre-declare std types, see http://www.gotw.ca/gotw/034.htm
+// We want to use `std::filesystem::path`, so we have it include it in the header.
+// <CODEGEN_COPY_TO_HEADER>
+
+#include <filesystem>
+
+// </CODEGEN_COPY_TO_HEADER>
 
 namespace rerun {
     namespace archetypes {
 
-#ifdef EDIT_EXTENSION
-        // [CODEGEN COPY TO HEADER START]
+#if 0
```

Comment:
Why not use `EDIT_EXTENSION` here?

Reply:
I'm phasing out the EDIT_EXTENSION way of doing things. The idea was to have LSP friendly code but that requires a bit of extra maintenance and everyone editing these (myself included) gets this wrong all the time. Also, it gave the impression that there's some additional magic at work.
So instead I make these being honest now and just explicitly disable stuff.